src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -202,6 +215,15 @@ public ZooKeeperSaslClient getSaslClient() {
 
     private final ZKWatchManager watchManager;
 
+    private ZKClientConfig clientConfig;
+
+    public ZKClientConfig getClientConfig() {
```

Comment:
@fpj @arshadmohammad Okay, in the javadoc of this class there's the following comment: "The methods of this class are thread-safe unless otherwise noted". Well... not this new one. 😈 

**First and foremost,** not being thread-safe **here** is not that **really** important/serious as we are basically retrieving the same config, but I only want to highlight that two threads can obtain __two different ZKClientConfig  objects__ by calling this method with the following interleaving:


         [clientConfig initially null]
         1. Thread-1 calls getConfigClient() method, jumps to line 221 and checks that clientConfig is null. 
          1.1. At about the same time Thread-2  calls getConfigClient(),
                and checks that clientConfig is also null (bad interleaving execution)
          2. The two threads call line 222 concurrently and the last one wins, that is, 
              one of threads overrides the value of the other one, so one of the clientConfig
             returned is an "orphaned" one (created but not assigned to clientConfig anymore)


Idk how much this is only a minor nuisance or a real problem (I am leaning towards the former, tbqh). 

To solve this, either do:

```
    public ZKClientConfig getClientConfig() {
        synchronized(this) {
            if (clientConfig == null) {
               clientConfig = new ZKClientConfig();
            }
        }
        return clientConfig;
    }
```

or ```public synchronized ZKClientConfig getClientConfig()```